### PR TITLE
build: support building with cmake

### PR DESCRIPTION
**Pull request by djphoenix**
Fixes: #1547

- Implement building and testing with cmake
- Download libsonic from repository (when enabled and not available as system library), ref: #650
- Migrate android native build to cmake
- Migrate CI to cmake
- Add windows CI

TODO (might be done in separate PR):
- Building as [cmake external project](https://cmake.org/cmake/help/latest/module/ExternalProject.html)
- Building for package managers (deb, rpm, ...)
- Strip out `vcproj` - _to be discussed_
- Strip out `autotools` after next release
- Update build documentation for `cmake`

### CODE REVIEW

File: tests/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+endmacro(shell_test)
+
+compiled_test(api)
+compiled_test(encoding)
+compiled_test(ieee80)
+compiled_test(readclause)
+
+if (SHELL AND UNIX)
+
+# shell_test(bom)
```
Comment (alex19EP) [inline tests/CMakeLists.txt:46]: why it's commented?

File: tests/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+# shell_test(non-executable-files-with-executable-bit)
+
+shell_test(cmd_options)
+shell_test(dictionary)
+shell_test(language-numbers-cardinal)
+shell_test(language-numbers-ordinal)
+# shell_test(language-phonemes)
```
Comment (alex19EP) [inline tests/CMakeLists.txt:53]: why it's commented?

**Review by alex19EP — CHANGES_REQUESTED** [review]

File: src/libespeak-ng/config.h.in [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+#cmakedefine01 USE_ASYNC
+#cmakedefine01 USE_KLATT
+#cmakedefine01 USE_LIBPCAUDIO
+#cmakedefine01 USE_LIBSONIC
+#cmakedefine01 USE_MBROLA
+#cmakedefine01 USE_SPEECHPLAYER
+
+#define PACKAGE "${PROJECT_NAME}"
+#define PACKAGE_NAME "${PROJECT_DESCRIPTION}"
```
[the remark this answers is not in the document]
Comment (djphoenix) [reply src/libespeak-ng/config.h.in:11]: That, at start, copies original autotools behavior, I tried to make `config.h` as near to original as possible. Indeed, old behavior (and these defines at all) is not used.

File: CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+add_subdirectory(src)
+include(cmake/data.cmake)
+
+include(cmake/package.cmake)
+include(CPack)
+
+include(CTest)
+add_subdirectory(tests)
+
+option(BUILD_SHARED_LIBS "Build shared libraries" OFF)
```
[the remark this answers is not in the document]
Comment (djphoenix) [reply CMakeLists.txt:22]: This is just an expose standard CMake variable to configuration utilities (like ccmake). By default, CMake behavior is to make static libraries. I think it anyway will be overridden by packaging tools, to make `.so` for libespeak-ng and `.h .a` for libespeak-ng-dev.

File: tests/CMakeLists.txt (hunk 2 above)
Comment (djphoenix) [reply tests/CMakeLists.txt:53]: That commented tests are not work inside separate build directory out of the box, I prepare nextup fixes to tests source that will bring it up again.

**Review by alex19EP — APPROVED** [review]
LGTM

**Pull request merged**